src/kmeans.jl

```diff
@@ -311,7 +317,9 @@ function update_centers!(
     # sum ==> mean
     @inbounds for j = 1:k
         if to_update[j]
-            centers[:, j] .*= 1 / cweights[j]
+            for i = 1:d
+                centers[i, j] /= cweights[j]
```

Comment:
I'm not 100% sure, but back in the days `/` was more expensive than `*`.

Reply:
Note that the current code computes `1/cweights[j]` only once, and then does the multiplication for each `i`. IIUC the timings for `*= 1 /` recompute the division it for each `i`, which isn't faster.

Reply:
Actually, I think LLVM is smart enough to calculate `1/wj` only once (at least that's what I see in `@code_llvm` for a toy example). But I still prefer the broadcast version (`cej = view(centers, :, j); cej .*= 1/cweights[j]`), where this would be definitely done once. :)

Reply:
I agree, the broadcasted version was nicer. Better use `view`. Same above, where `view` was used but you removed it: use `xj .* wj` and everything should be OK.